From c584447bc34b044727b50d0604b442824d90da36 Mon Sep 17 00:00:00 2001
From: Eddyhan <eddyhan@ami.com>
Date: Fri, 7 Oct 2022 10:45:22 +0800
Subject: [PATCH 2/2] create libspdm


diff --git a/library/spdm_common_lib/CMakeLists.txt b/library/spdm_common_lib/CMakeLists.txt
old mode 100644
new mode 100755
index 50cbd095..aed68962
--- a/library/spdm_common_lib/CMakeLists.txt
+++ b/library/spdm_common_lib/CMakeLists.txt
@@ -15,3 +15,5 @@ SET(src_spdm_common_lib
 )
 
 ADD_LIBRARY(spdm_common_lib STATIC ${src_spdm_common_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_common_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_crypt_lib/CMakeLists.txt b/library/spdm_crypt_lib/CMakeLists.txt
old mode 100644
new mode 100755
index b2325954..e59aecda
--- a/library/spdm_crypt_lib/CMakeLists.txt
+++ b/library/spdm_crypt_lib/CMakeLists.txt
@@ -9,3 +9,5 @@ SET(src_spdm_crypt_lib
 )
 
 ADD_LIBRARY(spdm_crypt_lib STATIC ${src_spdm_crypt_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_crypt_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_requester_lib/CMakeLists.txt b/library/spdm_requester_lib/CMakeLists.txt
old mode 100644
new mode 100755
index fa9bade3..8e46d3e3
--- a/library/spdm_requester_lib/CMakeLists.txt
+++ b/library/spdm_requester_lib/CMakeLists.txt
@@ -33,3 +33,5 @@ SET(src_spdm_requester_lib
 )
 
 ADD_LIBRARY(spdm_requester_lib STATIC ${src_spdm_requester_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_requester_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_responder_lib/CMakeLists.txt b/library/spdm_responder_lib/CMakeLists.txt
old mode 100644
new mode 100755
index c58d0b45..4d0b881d
--- a/library/spdm_responder_lib/CMakeLists.txt
+++ b/library/spdm_responder_lib/CMakeLists.txt
@@ -36,4 +36,6 @@ SET(src_spdm_responder_lib
     libspdm_rsp_chunk_get.c
 )
 
-ADD_LIBRARY(spdm_responder_lib STATIC ${src_spdm_responder_lib})
\ No newline at end of file
+ADD_LIBRARY(spdm_responder_lib STATIC ${src_spdm_responder_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_responder_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_secured_message_lib/CMakeLists.txt b/library/spdm_secured_message_lib/CMakeLists.txt
old mode 100644
new mode 100755
index c671b773..a3e23237
--- a/library/spdm_secured_message_lib/CMakeLists.txt
+++ b/library/spdm_secured_message_lib/CMakeLists.txt
@@ -12,3 +12,5 @@ SET(src_spdm_secured_message_lib
 )
 
 ADD_LIBRARY(spdm_secured_message_lib STATIC ${src_spdm_secured_message_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_secured_message_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_transport_mctp_lib/CMakeLists.txt b/library/spdm_transport_mctp_lib/CMakeLists.txt
old mode 100644
new mode 100755
index 3d42d058..95926e9b
--- a/library/spdm_transport_mctp_lib/CMakeLists.txt
+++ b/library/spdm_transport_mctp_lib/CMakeLists.txt
@@ -10,3 +10,5 @@ SET(src_spdm_transport_mctp_lib
 )
 
 ADD_LIBRARY(spdm_transport_mctp_lib STATIC ${src_spdm_transport_mctp_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_transport_mctp_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_transport_pcidoe_lib/CMakeLists.txt b/library/spdm_transport_pcidoe_lib/CMakeLists.txt
old mode 100644
new mode 100755
index 2e82cb5b..13d11766
--- a/library/spdm_transport_pcidoe_lib/CMakeLists.txt
+++ b/library/spdm_transport_pcidoe_lib/CMakeLists.txt
@@ -10,3 +10,5 @@ SET(src_spdm_transport_pcidoe_lib
 )
 
 ADD_LIBRARY(spdm_transport_pcidoe_lib STATIC ${src_spdm_transport_pcidoe_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_transport_pcidoe_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/cryptlib_openssl/CMakeLists.txt b/os_stub/cryptlib_openssl/CMakeLists.txt
old mode 100644
new mode 100755
index ee911849..501dab81
--- a/os_stub/cryptlib_openssl/CMakeLists.txt
+++ b/os_stub/cryptlib_openssl/CMakeLists.txt
@@ -38,3 +38,5 @@ SET(src_cryptlib_openssl
 
 ADD_LIBRARY(cryptlib_openssl STATIC ${src_cryptlib_openssl})
 TARGET_COMPILE_OPTIONS(cryptlib_openssl PRIVATE ${OPENSSL_FLAGS})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:cryptlib_openssl>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/debuglib/CMakeLists.txt b/os_stub/debuglib/CMakeLists.txt
old mode 100644
new mode 100755
index e2487b92..2e139331
--- a/os_stub/debuglib/CMakeLists.txt
+++ b/os_stub/debuglib/CMakeLists.txt
@@ -10,3 +10,7 @@ SET(src_debuglib
 )
 
 ADD_LIBRARY(debuglib STATIC ${src_debuglib})
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:debuglib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
+endif()
diff --git a/os_stub/debuglib_null/CMakeLists.txt b/os_stub/debuglib_null/CMakeLists.txt
index f0c1368f..7d291461 100644
--- a/os_stub/debuglib_null/CMakeLists.txt
+++ b/os_stub/debuglib_null/CMakeLists.txt
@@ -10,3 +10,7 @@ SET(src_debuglib_null
 )
 
 ADD_LIBRARY(debuglib_null STATIC ${src_debuglib_null})
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:debuglib_null>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
+endif()
diff --git a/os_stub/malloclib/CMakeLists.txt b/os_stub/malloclib/CMakeLists.txt
old mode 100644
new mode 100755
index d13a1607..2fd14cb0
--- a/os_stub/malloclib/CMakeLists.txt
+++ b/os_stub/malloclib/CMakeLists.txt
@@ -10,3 +10,5 @@ SET(src_malloclib
 )
 
 ADD_LIBRARY(malloclib STATIC ${src_malloclib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:malloclib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/memlib/CMakeLists.txt b/os_stub/memlib/CMakeLists.txt
old mode 100644
new mode 100755
index f2a7b481..f9522fdf
--- a/os_stub/memlib/CMakeLists.txt
+++ b/os_stub/memlib/CMakeLists.txt
@@ -13,3 +13,5 @@ SET(src_memlib
 )
 
 ADD_LIBRARY(memlib STATIC ${src_memlib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:memlib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/openssllib/CMakeLists.txt b/os_stub/openssllib/CMakeLists.txt
old mode 100644
new mode 100755
index 232a8b4f..b565483f
--- a/os_stub/openssllib/CMakeLists.txt
+++ b/os_stub/openssllib/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.8.12)
 
-ADD_COMPILE_OPTIONS(-DLIBSPDM_OPENSSL_STDINT_WORKAROUND -DL_ENDIAN -DOPENSSL_SMALL_FOOTPRINT -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -DOPENSSL_SYS_UEFI=1 ${OPENSSL_FLAGS})
+ADD_COMPILE_OPTIONS(-DLIBSPDM_OPENSSL_STDINT_WORKAROUND -DL_ENDIAN -DOPENSSL_SMALL_FOOTPRINT -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -DOPENSSL_SYS_UEFI=1 -Wno-error=array-parameter= ${OPENSSL_FLAGS})
 
 INCLUDE_DIRECTORIES(${LIBSPDM_DIR}/include
                     ${LIBSPDM_DIR}/include/hal 
@@ -545,3 +545,5 @@ SET(src_openssllib
 )
 
 ADD_LIBRARY(openssllib STATIC ${src_openssllib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:openssllib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/platform_lib/CMakeLists.txt b/os_stub/platform_lib/CMakeLists.txt
old mode 100644
new mode 100755
index 7fc766c1..cc827d22
--- a/os_stub/platform_lib/CMakeLists.txt
+++ b/os_stub/platform_lib/CMakeLists.txt
@@ -18,3 +18,5 @@ SET(src_platform_lib
 endif()
 
 ADD_LIBRARY(platform_lib STATIC ${src_platform_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:platform_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/rnglib/CMakeLists.txt b/os_stub/rnglib/CMakeLists.txt
old mode 100644
new mode 100755
index 360ced88..55a12bf1
--- a/os_stub/rnglib/CMakeLists.txt
+++ b/os_stub/rnglib/CMakeLists.txt
@@ -20,3 +20,5 @@ SET(src_rnglib
 endif()
 
 ADD_LIBRARY(rnglib STATIC ${src_rnglib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:rnglib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt b/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
old mode 100644
new mode 100755
index fcb483bb..3b8a5743
--- a/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
+++ b/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
@@ -12,3 +12,5 @@ SET(src_spdm_device_secret_lib_sample
 )
 
 ADD_LIBRARY(spdm_device_secret_lib_sample STATIC ${src_spdm_device_secret_lib_sample})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_device_secret_lib_sample>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
-- 
2.34.1

