From 094423073504c23b3f3a47e34458a421a0e70b76 Mon Sep 17 00:00:00 2001
From: yihuangx <yi-huangx.han@intel.com>
Date: Wed, 23 Mar 2022 23:10:14 -0700
Subject: [PATCH] Disable Node Manager when Partition Boot is enabled.

Upstream-Status: Pending
---
 xyz.openbmc_project.NodeManager.service | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xyz.openbmc_project.NodeManager.service b/xyz.openbmc_project.NodeManager.service
index 5a96bce..0dfb772 100644
--- a/xyz.openbmc_project.NodeManager.service
+++ b/xyz.openbmc_project.NodeManager.service
@@ -3,6 +3,8 @@ Description=Node Manager Service
 Wants=dbus.service ipmb.service xyz.openbmc_project.intelcpusensor.service nv-sync.service
 After=dbus.service ipmb.service xyz.openbmc_project.intelcpusensor.service nv-sync.service
 
+ConditionPathExists=!/var/cache/private/PartitionBootFlag
+
 [Service]
 ExecStart=/usr/sbin/node-manager
 Restart=on-failure
-- 
2.25.1

